subprojects/core/src/main/java/org/gradle/api/tasks/options/Option.java

```diff
@@ -29,15 +29,14 @@
  *
  * <p>An option may have one of the following types:</p>
  * <ul>
- * <li>{@code boolean}</li>
- * <li>{@code Boolean}</li>
- * <li>{@code Property<Boolean>}</li>
- * <li>an {@code enum} type</li>
- * <li>{@code Property<T>} of an enum type</li>
- * <li>{@code String}</li>
- * <li>{@code Property<String>}</li>
- * <li>{@code List<T>} of an {@code enum} type</li>
- * <li>{@code List<String>}</li>
+ * <li>{@code boolean}, {@code Boolean}, or {@code Property<Boolean>}</li>
+ * <li>{@code Integer} or {@code Property<Integer>}</li>
```

Comment:
:x: We should at least support `Long` and `Double` as well. I think it's fine to skip `Short` and `Byte` and `Float` as they are rarely supported and can be substituted with `Integer` and `Double` easily.